pkg/api/operators/operators_test.go

```diff
@@ -0,0 +1,266 @@
+			// append WLS operator
+			createAndInsertWLSOperator(1, tt)
+			// setup http request processing
+			recorder := httptest.NewRecorder()
+			router := mux.NewRouter()
+			router.HandleFunc("/operators/{id}", func(w http.ResponseWriter, r *http.Request) {
+				ReturnSingleOperator(w, r)
+			})
+			request, _ := http.NewRequest("GET", "/operators/1", nil)
+			// inovke HTTP request
+			router.ServeHTTP(recorder, request)
+			// assert response code 200
+			assert.Equal(t, http.StatusOK, recorder.Code, "wrong response code")
+			var op Operator
+			err := json.Unmarshal(recorder.Body.Bytes(), &op)
+			if err != nil {
+				t.Errorf("Error while reading response JSON: %s", err)
+			}
+			// Validate the correct number of operators (1), and the operator has the expected name
+			assert.NotNil(t, op, "Operator should not be nil")
+			assert.Equal(t, "wls-operator-mytestbinding-1", op.Name, "Operator has the wrong name")
+		})
+	}
+}
+
+//TestReturnAllOperators ensures that an array of operators is returned
+func TestReturnAllOperators(t *testing.T) {
+	// Get a set of clusters
+	Operators = nil
+	clusters := testutil.GetTestClusters()
+	var clients kubernetes.Interface = fake.NewSimpleClientset()
+	modelBindingPairs := map[string]*types.ModelBindingPair{
+		"test-pair-1": testutil.GetModelBindingPair(),
+	}
+	type args struct {
+		listers controller.Listers
+	}
+	tests := []struct {
+		name    string
+		listers controller.Listers
+	}{
+		{name: "testReturnAllOperators",
+			listers: testutilcontroller.NewControllerListers(&clients, clusters, &modelBindingPairs)},
+	}
+	for _, tt := range tests {
+		// add a WLSOperator to one of the clusters
+		t.Run(tt.name, func(t *testing.T) {
+			// append the WLS Operator
+			createAndInsertWLSOperator(1, tt)
+			createAndInsertWLSOperator(2, tt)
+			// setup http request processing
+			recorder := httptest.NewRecorder()
+			router := mux.NewRouter()
+			router.HandleFunc("/operators", func(w http.ResponseWriter, r *http.Request) {
+				ReturnAllOperators(w, r)
+			})
+			request, _ := http.NewRequest("GET", "/operators", nil)
+			// inovke HTTP request
+			router.ServeHTTP(recorder, request)
+			// assert response code 200
+			assert.Equal(t, http.StatusOK, recorder.Code, "wrong response code")
+			var ops []Operator
+			err := json.Unmarshal(recorder.Body.Bytes(), &ops)
+			if err != nil {
+				t.Errorf("Error while reading response JSON: %s", err)
+			}
+			// Validate that there is the correct number of operators (1), and the
+			// operator has the expected name
+			assert.Equal(t, 2, len(ops), "There should be one operator in the list")
+			assert.Equal(t, "wls-operator-mytestbinding-1", ops[0].Name, "Operator has the wrong name")
+			assert.Equal(t, "wls-operator-mytestbinding-2", ops[1].Name, "Operator has the wrong name")
+		})
+	}
+}
+
+//TestReturn404ForMissingOperator should return a 404 response for a request for a non-existent operator
+func TestReturn404ForMissingOperator(t *testing.T) {
```

Comment:
If you're so inclined, you can combine the 3 ReturnSingleOperator cases into a single test method with the 3 cases spelled out. It is a matter of personal preference so feel free to ignore this comment